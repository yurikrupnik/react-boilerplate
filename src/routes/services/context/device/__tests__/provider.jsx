import React from 'react';
import { Consumer, Provider } from '../index';
import themeConfig from '../../../../../theme';

const {
    describe,
    test,
    expect,
    shallow,
    render,
    navigator
} = global;

describe('Device context', () => {
    const Wrapper = () => ((
        <div>hello</div>
    ));
    test('render consumer', () => {
        const wrapper = shallow( // notice render here
            <Consumer render={() => (
                <Wrapper />
            )}
            />
        );
        expect(wrapper).toMatchSnapshot();
    });
    test('render provider', () => {
        const wrapper = shallow((
            <Provider theme={themeConfig} userAgent={navigator.userAgent}>
                <Wrapper />
            </Provider>
        ));
        expect(wrapper).toMatchSnapshot();
    });
    test('defualt props', () => {
        const wrapper = shallow((
            <Provider theme={themeConfig} userAgent={navigator.userAgent}>
                <Wrapper />
            </Provider>
        ));
        expect(wrapper).toMatchSnapshot();
    });

    test('methods', () => {
        const wrapper = render((
            <Provider theme={themeConfig} userAgent={navigator.userAgent}>
                <Consumer render={(props) => {
                    const { isMobile, toggle } = props;
                    toggle();
                    toggle();
                    // isMobile();
                    return (
                        <Wrapper />
                    );
                }}
                />
            </Provider>
        ));
        expect(wrapper).toMatchSnapshot();
    });
});
