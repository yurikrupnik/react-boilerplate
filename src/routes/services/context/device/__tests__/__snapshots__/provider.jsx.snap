// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Device context defualt props 1`] = `
<ContextProvider
  value={
    Object {
      "browser": "",
      "engine": "Webkit",
      "isMobile": [Function],
      "os": "Windows",
      "toggle": [Function],
      "type": "Desktop",
      "userAgent": "Mozilla/5.0 (darwin) AppleWebKit/537.36 (KHTML, like Gecko) jsdom/11.12.0",
      "version": "5.0",
    }
  }
>
  <Wrapper />
</ContextProvider>
`;

exports[`Device context methods 1`] = `
<div>
  hello
</div>
`;

exports[`Device context render consumer 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`Device context render provider 1`] = `
<ContextProvider
  value={
    Object {
      "browser": "",
      "engine": "Webkit",
      "isMobile": [Function],
      "os": "Windows",
      "toggle": [Function],
      "type": "Desktop",
      "userAgent": "Mozilla/5.0 (darwin) AppleWebKit/537.36 (KHTML, like Gecko) jsdom/11.12.0",
      "version": "5.0",
    }
  }
>
  <Wrapper />
</ContextProvider>
`;
